### PR TITLE
iOS attributable_clients use metrics adclicks

### PR DESCRIPTION
**Pull request by fbertsch**
Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-1950)

### CODE REVIEW

File: sql/moz-fx-data-shared-prod/firefox_ios_derived/attributable_clients_v1/query.sql [hunk 1]
```diff
@@ -3,23 +3,52 @@ RETURNS INT64 AS (
   (SELECT SUM(value) FROM UNNEST(map))
 );
 
-WITH client_search_activity AS (
+WITH client_days AS (
   SELECT
     DATE(submission_timestamp) AS submission_date,
     sample_id,
     client_info.client_id,
+    LOGICAL_OR(mozfun.norm.extract_version(app_version, 'major') >= 108) AS has_search_data,
```
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/firefox_ios_derived/attributable_clients_v1/query.sql:11]: How does extracting the app_version and checking it guarantees there is search data?

File: sql/moz-fx-data-shared-prod/firefox_ios_derived/attributable_clients_v1/query.sql [hunk 2]
```diff
@@ -3,23 +3,52 @@ RETURNS INT64 AS (
...
     SUM(sum_map_values(metrics.labeled_counter.search_in_content)) AS searches,
     SUM(sum_map_values(metrics.labeled_counter.browser_search_with_ads)) AS searches_with_ads,
     SUM(sum_map_values(metrics.labeled_counter.browser_search_ad_clicks)) AS ad_clicks,
   FROM
     firefox_ios.baseline
   WHERE
-    DATE(submission_timestamp) = @submission_date
+    {% if is_init() %}
+      submission_date >= "2020-05-01"
```
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/firefox_ios_derived/attributable_clients_v1/query.sql:19]: Why May and not January?

File: sql/moz-fx-data-shared-prod/firefox_ios_derived/attributable_clients_v1/query.sql [hunk 3]
```diff
@@ -3,23 +3,52 @@ RETURNS INT64 AS (
...
+  SELECT
+    submission_date,
+    client_id,
+    sample_id,
+    LOGICAL_OR(mozfun.norm.extract_version(app_version, 'major') < 108) AS has_search_data,
+    SUM(search_count) AS searches,
+    SUM(search_with_ads) AS searches_with_ads,
+    SUM(ad_click) AS ad_clicks
+  FROM
+    `moz-fx-data-shared-prod.search_derived.mobile_search_clients_daily_v1`
```
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/firefox_ios_derived/attributable_clients_v1/query.sql:38]:
```suggestion
    search_derived.mobile_search_clients_daily_v1
```

**Review by kik-kik — APPROVED** [review]
r+wc

File: sql/moz-fx-data-shared-prod/firefox_ios_derived/attributable_clients_v1/query.sql (hunk 1 above)
Comment (fbertsch) [reply sql/moz-fx-data-shared-prod/firefox_ios_derived/attributable_clients_v1/query.sql:11]: Past version `108` on release, we were reporting search data on the baseline ping. Prior to 108, we were reporting it just on the metrics ping. This kind of check also handles clients who haven't upgraded yet: they will report e.g. `107` for the search values, so we will only get the search data from the metrics ping.

File: sql/moz-fx-data-shared-prod/firefox_ios_derived/attributable_clients_v1/query.sql (hunk 2 above)
Comment (fbertsch) [reply sql/moz-fx-data-shared-prod/firefox_ios_derived/attributable_clients_v1/query.sql:19]: Data started being reported for `firefox_ios.baseline` in May. I can add a link to a query

**Comment by fbertsch** [comment]
Thanks for the review @kik-kik!

**Pull request merged**